modules/rest-api/src/main/java/org/apache/ignite/internal/rest/api/recovery/system/ResetClusterRequest.java

```diff
@@ -25,28 +25,48 @@
 import java.util.Objects;
 import org.apache.ignite.internal.tostring.IgniteToStringInclude;
 import org.apache.ignite.internal.tostring.S;
+import org.jetbrains.annotations.Nullable;
 
 /** Request to reset cluster. */
 @Schema(description = "Reset cluster.")
 public class ResetClusterRequest {
-    @Schema(description = "Names of the proposed CMG nodes.")
+    @Schema(description = "Names of the proposed CMG nodes. Optional if Metastorage replication factor is specified, then "
+            + "current CMG nodes will be used.")
     @IgniteToStringInclude
-    private final List<String> cmgNodeNames;
+    private final @Nullable List<String> cmgNodeNames;
+
+    @Schema(description = "Number of nodes in the Raft voting member set for Metastorage.")
+    @IgniteToStringInclude
+    private final @Nullable Integer metastorageReplicationFactor;
 
     /** Constructor. */
     @JsonCreator
-    public ResetClusterRequest(@JsonProperty("cmgNodeNames") List<String> cmgNodeNames) {
-        Objects.requireNonNull(cmgNodeNames);
+    public ResetClusterRequest(
+            @JsonProperty("cmgNodeNames") @Nullable List<String> cmgNodeNames,
+            @JsonProperty("metastorageReplicationFactor") @Nullable Integer metastorageReplicationFactor
+    ) {
+        if (metastorageReplicationFactor == null) {
+            Objects.requireNonNull(cmgNodeNames);
```

Comment:
What will happen if the user does not send any of these fields? It seems that this will result in an NPE. Let's remove this `requireNonNull()` and make the validation in the manager (throwing a `ClusterResetException`)